heka/sandbox/filters/firefox_executive_report.lua

```diff
@@ -0,0 +1,237 @@
+    local _os     = get_os_id()
+
+    local partition = format("%d,%d,%d", country, channel, _os)
+    local r = interval[partition]
+    if not r then
+        local time_str = date(time_t_fmt, time_t)
+        -- date, actives, hours, inactives, new_records, five_of_seven, total_records, crashes, default, google, bing, yahoo, other
+        r = {time_str, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
+        interval[partition] = r
+    end
+    return r, country, channel, _os
+end
+
+
+local function update_row(r, cid, country, channel, _os, dow)
+    local dflt = fx.get_boolean_value(read_message("Fields[default]"))
+    fx_cids:add(cid, country, channel, _os, dow, dflt)
+
+    local doc_type = read_message("Fields[docType]")
+    if doc_type == "main" then
+        r[3]  = r[3]  + (tonumber(read_message("Fields[hours]")) or 0)
+        r[10] = r[10] + (tonumber(read_message("Fields[google]")) or 0)
+        r[11] = r[11] + (tonumber(read_message("Fields[bing]")) or 0)
+        r[12] = r[12] + (tonumber(read_message("Fields[yahoo]")) or 0)
+        r[13] = r[13] + (tonumber(read_message("Fields[other]")) or 0)
+    elseif doc_type == "crash" then
+        r[8] = r[8] + 1
+    end
+end
+
+
+local function clear_intervals(s, e, size)
+    for i = s + 1, e do
+        local idx = i % size + 1
+        intervals[idx] = {}
+    end
+    current_interval = e
+end
+
+
+local function update_day(ts, cid, day)
+    if current_interval == -1 then current_interval = day end
+
+    local delta = day - current_interval
+    if delta > 0 and delta < DAYS then
+        fx_cids:report(intervals[current_interval % DAYS + 1])
+        clear_intervals(current_interval, day, DAYS)
+    elseif delta >= DAYS then
+        error(string.format("data gap over %d days", DAYS))
+    end
+
+    local t = intervals[day % DAYS + 1]
+    local r, country, channel, _os = get_row(t, "%Y-%m-%d", ts / 1e9)
+    if r then
+        update_row(r, cid, country, channel, _os, 0)
+    end
+end
+
+
+local function update_week(ts, cid, day)
+    local week = floor((day + DAY_OFFSET) / 7)
+    if current_interval == -1 then current_interval = week end
+
+    local delta = week - current_interval
+    if delta > 0 and delta < WEEKS then
+        fx_cids:report(intervals[current_interval % WEEKS + 1])
+        clear_intervals(current_interval, week, WEEKS)
+    elseif delta >= WEEKS then
+        error(string.format("data gap over %d weeks", WEEKS))
+    end
+
+    local interval = intervals[week % WEEKS + 1]
+    local r, country, channel, _os = get_row(interval, "%Y-%m-%d", week * SEC_IN_WEEK - (DAY_OFFSET * SEC_IN_DAY))
+    if r then
+        -- The use of submission date changes the meaning of the day of the week
+        -- calculation, it now represents the days the user interacted with the
+        -- telemetry system.
+        update_row(r, cid, country, channel, _os, (day + DAY_OFFSET) % 7)
+    end
+end
+
+
+local function update_month(ts, cid, day, day_changed)
```

Comment:
Never mind... I see it's for compatibility with other "update_interval" functions.